### PR TITLE
update PR tempelate with more testing check boxes

### PR DESCRIPTION
**Pull request by ramezsaeed**
### Description

Describe the changes introduced by this PR and what does it affect

### Changes

List of changes this PR includes

### Related Issues

List of related issues

### Tested Scenarios

A list of scenarios tried to match the deliverables

### Documentation PR

For UI changes, Please provide the Documetation PR on [info_grid](https://github.com/threefoldtech/info_grid)

#### Preliminary Checks:

- Does it completely address the issue linked?
- What about edge cases?
- Does it meet the specified acceptance criteria?
- Are there any unintended side effects?
- Does the PR adhere to the team's coding conventions, style guides, and best practices?
- Does it integrate well with existing features?
- Does it impact the overall performance of the application?
- Are there any bottlenecks or slowdowns?
- Has it been optimized for efficiency?
- Has it been adequately tested with unit, integration, and end-to-end tests?
- Are there any known defects or issues?
- Is the code well-documented?
- Are changes to documentation reflected in the code?

#### UI Checks:

- If a UI design is provided/ does it follow it?
- Does every button work?
- Is the data displayed logical? Is it what you expected?
- Does every validation work?
- Does this interface feel intuitive?
- What about slow network? Offline?
- What if the gridproxy/graphql/chain is failing?
- What would a first time user have a hard time navigating here?

#### Code Quality Checks:

- Code formatted/linted? Are there unused imports? .. etc
- Is there redundant/repeated code?
- Are there conditionals that are always true or always false?
- Can we write this more concisely?
- Can we reuse this code? If yes, where?
- Will the changes be easy to maintain and update in the future?
- Can this code become too complex to understand for other devs?
- Can this code cause future integration problems?

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -58,6 +58,16 @@ For UI changes, Please provide the Documetation PR on [info_grid](https://github
 - Can this code become too complex to understand for other devs?
 - Can this code cause future integration problems?
 
+### Testing Checklist
+
+- [ ] Meet the specified acceptance criteria.
+- [ ] Test if can affect any other functionality.
+- [ ] Tested with unit, integration, and end-to-end tests.
+- [ ] Tested the un-happy path and Rollback scenarios.
+- [ ] Documentation updated to meat the latest changes.
```
Comment (xmonader) [inline .github/PULL_REQUEST_TEMPLATE.md]: meet*

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -58,6 +58,16 @@ For UI changes, Please provide the Documetation PR on [info_grid](https://github
 - Can this code become too complex to understand for other devs?
 - Can this code cause future integration problems?
 
+### Testing Checklist
+
+- [ ] Meet the specified acceptance criteria.
+- [ ] Test if can affect any other functionality.
```
Comment (amiraabouhadid) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Test if changes can affect any other functionality.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -58,6 +58,16 @@ For UI changes, Please provide the Documetation PR on [info_grid](https://github
 - Can this code become too complex to understand for other devs?
 - Can this code cause future integration problems?
 
+### Testing Checklist
+
+- [ ] Meet the specified acceptance criteria.
```
Comment (amiraabouhadid) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Meets the specified acceptance criteria.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -58,6 +58,16 @@ For UI changes, Please provide the Documetation PR on [info_grid](https://github
...
+- [ ] Test if can affect any other functionality.
+- [ ] Tested with unit, integration, and end-to-end tests.
+- [ ] Tested the un-happy path and Rollback scenarios.
```
Comment (amiraabouhadid) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Tested the un-happy path and rollback scenarios.
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (ramezsaeed) [reply .github/PULL_REQUEST_TEMPLATE.md:67]: copy paste lelasaf :D

will fix the line copied from too :D

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/2f2d755f8845d0af2897450dc4b9d5f3d4c35eb5/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L27

**Review by amiraabouhadid — APPROVED** [review]

**Review by xmonader — CHANGES_REQUESTED** [review]
I don't agree with this 

I was expecting it be a grouped check list e.g 

image


Now none of the items are really considered.

**Comment by ramezsaeed** [comment]
> I don't agree with this
> 
> I was expecting it be a grouped check list e.g
> 
> image
> 
> Now none of the items are really considered.

OKey
will do that grouping.

I just take what is related to testing now and make them in a separate check boxes.

so what you need now is only make them in groups and remove the section I had added ?

**Comment by xmonader** [comment]
You can Add more, but the PR author needs to be always reminded of what they need to check

**Comment by maayarosama** [comment]
codacy check is failling please fix it

**Comment by ramezsaeed** [comment]
> codacy check is failling please fix it

fixed

**Review by maayarosama — APPROVED** [review]

**Pull request merged**